oap-server/server-query-plugin/query-graphql-plugin/src/main/java/org/apache/skywalking/oap/query/graphql/resolver/AlarmQuery.java

```diff
@@ -62,11 +83,111 @@ public Alarms getAlarm(final Duration duration, final Scope scope, final String
         }
         long startSecondTB = 0;
         long endSecondTB = 0;
+        EventQueryCondition condition = new EventQueryCondition();
         if (nonNull(duration)) {
             startSecondTB = duration.getStartTimeBucketInSec();
             endSecondTB = duration.getEndTimeBucketInSec();
+            condition.setTime(duration);
+        }
+        Alarms alarms = getQueryService().getAlarm(
+                scopeId, keyword, paging, startSecondTB, endSecondTB, tags);
+        Events events = null;
+        try {
+            events = getEventQueryService().queryEvents(condition);
```

Comment:
So, you only get to query a maximum of 100 events. SkyWalking could easily face 300+ services, 10k+instances, and countless endpoints (even we don't recommend alarm on the endpoint). This seems not the right call.

Reply:
I think we should the alarm list first, and query the event accordingly, @kezhenxu94 What do you think?
If you have concerns about performance, try concurrently query.

Reply:
> I think we should the alarm list first, and query the event accordingly, @kezhenxu94 What do you think?
> If you have concerns about performance, try concurrently query.

+1, please narrow the `condition` as much as possible, querying events by time range may cause critical performance problem

Reply:
Also, I don't think we want to include the `Alarm` events in the alarm messages, it seems to be pointless to me.

Reply:
Ok. Thanks.

Reply:
> You were proposing the query protocol, with the event included in the alarm. Why are you asking me about this? If you are thinking about 2 steps query, why we haven't discussed this at the protocol level.

Sorry, it was only after I received the request to narrow the scope of the query that I looked at the code and found that the current single-step query mode seemed unable to construct the query condition completely(The query criteria only have the **Scope Type**, without the  **Source values**.).
That's why I brought up the topic of a two-step query. I would like to confirm with you that we may need to roll back the query-protocol to the previous version. Sorry for my mistake.

if there are some problems in my way of asking questions, thanks for helping me to point out. I just want to do something to make SkyWalking to be better.Thanks for your reply.I'll pay attention next time.